ledger/participant-integration-api/src/main/resources/db/migration/h2database-appendonly/V1__Append_only_schema.sql

```diff
@@ -117,11 +117,25 @@ CREATE TABLE participant_command_submissions (
 CREATE TABLE participant_command_completions (
     completion_offset VARCHAR NOT NULL,
     record_time TIMESTAMP NOT NULL,
-    application_id VARCHAR NOT NULL,
+    -- The application ID has to be provided by the application.
+    -- Nullable for alignment with the PostgreSQL schema.
+    application_id VARCHAR,
     submitters ARRAY NOT NULL,
     command_id VARCHAR NOT NULL,
     -- The transaction ID is `NULL` for rejected transactions.
     transaction_id VARCHAR,
+    -- The submission ID will be provided by the participant or driver if the application didn't provide one.
+    -- Nullable to support historical data.
+    submission_id VARCHAR,
+    -- The three alternatives below are mutually exclusive, i.e. the deduplication
+    -- interval could have specified by the application as one of:
+    -- 1. an initial offset
+    -- 2. a duration (split into two columns, seconds and nanos, mapping protobuf's 1:1)
+    -- 3. an initial timestamp
+    deduplication_offset VARCHAR,
+    deduplication_time_seconds BIGINT,
+    deduplication_time_nanos INT,
+    deduplication_start TIMESTAMP,
```

Comment:
This way of specifying the deduplication period is currently unsupported but it belongs here for imminent use, unsure if removing it or leaving it for posterity.

Reply:
I agree with leaving it and keeping the migration count down.